src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -989,6 +989,14 @@ internal override void UpdateThemeBindings(Data.ResourceUpdateReason updateReaso
 			base.UpdateThemeBindings(updateReason);
 
 			SetDefaultForeground(ForegroundProperty);
+
+			if (_inlines is not null)
+			{
+				foreach (var inline in _inlines)
+				{
+					((IDependencyObjectStoreProvider)inline).Store.UpdateResourceBindings(updateReason);
+				}
+			}
```

Comment:
I think this should be the responsibility of DependencyObjectStore to do this. This is mostly a workaround to avoid changing DependencyObjectStore for now.